### DISCUSSION TITLE
Wiki Ideas, Suggestions, Feedback

### DISCUSSION THREAD

**Discussion by ThioJoe** [ThioJoe/YT-Spammer-Purge#303] [General]
I just wrote up a whole bunch of documentation for the Wiki: https://github.com/ThioJoe/YT-Spammer-Purge/wiki

Feel free to make any suggestions on what additional topics and info might be worth me adding.

**Comment by Kurdakov** [comment]
some people under your video ask how to install on Android,
I wrote something here https://github.com/ThioJoe/YT-Spammer-Purge/discussions/279#discussioncomment-1891015
but that needs to be verified and if correct might be added as installation instructions on Android (Android itself is not very utility friendly - for example it is difficult to get spammer id (more steps needed than on desktop systems)), but if you are ok with bunch of questions about Android installation/use problems - then Android installation instructions might have a right to be in Wiki

**Reply by Kurdakov** [reply]
There could be mention of https://github.com/microsoft/terminal 
on Windows 10 it could be run in python mode with all emojis 
on Windows 11 it is possible to set Windows Terminal as default console https://devblogs.microsoft.com/commandline/windows-terminal-preview-1-9-release/ (Default terminal) , so exe will run in Windows Terminal when started (on Windows 10 it is also possible but only on insider builds - so possibly will be available in first 2022 big update )

**Reply by rachmadaniHaryono** [reply]
maybe use termux https://termux.com/


i don't have any experience with it but i assume it require the program to be packageable first

**Comment by kiatchun** [comment]
Add screenshots to the wiki.

**Comment by KendallDoesCoding** [comment]
> I just wrote up a whole bunch of documentation for the Wiki: https://github.com/ThioJoe/YT-Spammer-Purge/wiki
> 
> Feel free to make any suggestions on what additional topics and info might be worth me adding.

@ThioJoe This is good and all, but here you can only do the edits in wiki. I suggest you use docsify like I mentioned in #300 , you can even make another repo and use docsify if you dont want to add it into this repo.

**Reply by rbeer** [reply]
Just create a /docs folder in the repo and put the .md files that would make up the wiki pages in there.
GitHub renders .md the same way, no matter where they are. Wiki pages are markdown files.

`git clone https://github.com/ThioJoe/YT-Spammer-Purge.wiki` - No you can't PR from the wiki repo :/

**Comment by KendallDoesCoding** [comment]
Yeah, this would be a better way lol

**Comment by Kurdakov** [comment]
When I started to see heavily infested channels, I thought that owners should have separate entry in wiki.

When they first run utility and there are thousands of spam posts - there is potential to ban dozens of accounts  which just use too much emoji but which are not spammers.

of cause, experienced user would just create few client secrets, then will run and save log, exiting by Ctrl-C when action is offered, then extracting ids from log, delete most of scammers by ids, then run again in detection mode, to see what is left.

so for beginners from heavily infested channels 
maybe something along the lines  (in quotes something which I thing should be in wiki)

"If you are owner of heavily infested channel and want to clean it.
Be prepared to be patient. first run- save log and exit program Ctrl-C when it offers to delete hold or report
now - run program again deleting  by  ids (for correct spam messages) from  log.
now after that - run just as if spam is infrequent." 

maybe  option just "save log for analysis and exit" (even though log is saved before action operation)  will be useful in future in utility.

"if quota is exceeded - run with another client secret (fortunately it's possible to create few projects in google dev console)"

for future possibly there could be non api run (via http requests)  which will only creates log, 
then log might be manually cleaned, then batch processed with all left ids with selected operation (hold, remove) with api calls.

**Reply by ThioJoe** [reply]
Yea I'm going to add a way to delete a list of comments using a log file, sort of like how it's done with recovery files.

Also, the number of emojis has no effect on flagging it as spam

**Comment by Kurdakov** [comment]
A user closed issue https://github.com/ThioJoe/YT-Spammer-Purge/issues/359 with question on options scan modes 2 and 3 for non owners and non moderators. It appears currently the only option for non owners is  to report 
via 1 st option to scan ( in option 3 when IDs are entered reports  are also not allowed ).

Somehow either IDs could be allowed to report or it should be clearly stated that non channel owners could only use only 1 st option (Auto Smart scan mode) - at least there is no clear direction in docs on that issue.

Also - disabled options might be colored somehow to non channel owners (greyed?)

**Reply by ThioJoe** [reply]
Yea that's not a bad idea to grey-out, but not forbid those options

**Comment by resucutie** [comment]
Add the current default value from each setting in the [Config File Settings](https://github.com/ThioJoe/YT-Spammer-Purge/wiki/Config-File-Settings)

**Comment by TechStudent10** [comment]
A GitHub Pages site maybe?

**Reply by rachmadaniHaryono** [reply]
there is this discussion on using docsify https://github.com/ThioJoe/YT-Spammer-Purge/discussions/300

but i use some parts from https://github.com/ThioJoe/YT-Spammer-Purge/pull/489 from @dekoza, which is generated from https://github.com/pawamoy/copier-poetry

https://rachmadaniharyono.github.io/YT-Spammer-Purge/

~~i will write complete description later~~

**Reply by rachmadaniHaryono** [reply]
## result 

https://rachmadaniharyono.github.io/YT-Spammer-Purge/

https://github.com/rachmadaniHaryono/YT-Spammer-Purge/tree/feature/mkdocs

some of this is from pr #489 by @dekoza

## diff

to view diff #489 and https://github.com/rachmadaniHaryono/YT-Spammer-Purge/tree/feature/mkdocs run following script


```shell
> git diff dekoza/rewrite -- \
. \
':(exclude).github' \
':(exclude).gitignore' \
':(exclude).pre-commit-config.yaml' \
':(exclude)assets' \
':(exclude)CHANGELOG.md' \
':(exclude)CODE_OF_CONDUCT.md' \
':(exclude)config' \
':(exclude)CONTRIBUTING.md' \
':(exclude)docker' \
':(exclude)docker-compose.override.yml' \
':(exclude)docker-compose.yml' \
':(exclude)Dockerfile' \
':(exclude)duties.py' \
':(exclude)install.sh' \
':(exclude)install_full.sh' \
':(exclude)Makefile' \
':(exclude)OLD_README.md' \
':(exclude)README.md' \
':(exclude)requirements.txt' \
':(exclude)Scripts' \
':(exclude)src' \
':(exclude)tests' \
':(exclude)YTSpammerPurge.py' 
```

- `docs/gen_rev_nav.py` is based on https://github.com/mkdocstrings/mkdocstrings/blob/master/docs/gen_ref_nav.py
- `docs/macros.py` is little bit different to black formatter
- `docs/reference/cli.md` because the program don't have cli module
- `mkdocs.yml`
  - it is formatted with https://github.com/macisamuele/language-formatters-pre-commit-hooks
  - few file are not included
    - code of conduct related files, see #103
    - changelog because there is no changelog file
    - coverage because there is no test
  - mkdocstrings watch configuration changed based on current structure
  - added `literate-nav` for generated `SUMMARY.md`
- pyproject
  - little bit different but it is more similar to #459
  - package config have to be modified based on current structure
- mkdocstrings config
  - option `show_if_no_docstring` is `true` because mkdocstrings can't catch any docstring
  - option `separate_signature` is `true` because it is rendered better on heading

## github page and wiki

currently there is no way to get wiki on mkdocs, see

https://github.com/mkdocs/mkdocs/issues/1437

how it is currently done

- clone wiki to folder under docs

`git clone https://github.com/ThioJoe/YT-Spammer-Purge.wiki.git docs/wiki`

- add that folder to `mkdocs.yml`'s nav

it mean you have to run `git pull` on sub repo to get latest wiki edit everytime you publish github page

pros: there is no split of documentation between wiki and github page

cons:

- all link that refer other wiki page will link to wiki page instead of github page wiki section
- some page rendered differently especially `_Footer.md`
- wiki page is sorted alphabetically than by hierarchy
- because above point first page on wiki section is `Checking For Updates` instead of `Home`

## how to run it

get the branch

https://github.com/rachmadaniHaryono/YT-Spammer-Purge/tree/feature/mkdocs

run `poetry run mkdocs serve` to run live server

run `poetry run mkdocs gh-deploy` to deploy github page
